pkg/dogstatsd/server.go

```diff
@@ -299,6 +319,16 @@ func (s *Server) parsePacket(packet *listeners.Packet, metricSamples []*metrics.
 				dogstatsdMetricParseErrors.Add(1)
 				continue
 			}
+
+			if s.mapper != nil && len(sample.Tags) == 0 {
+				name, tags, matched := s.mapper.GetMapping(sample.Name)
+
+				if matched {
+					sample.Name = name
+					sample.Tags = append(sample.Tags, tags...)
```

Comment:
Nit: you could probably remove a few allocs by pre-building `sample.Tags` to the same size as `tags`. Probably not worth since I don't think we expect very high traffic through this path ?

Reply:
Let's wait #4427 :)

Reply:
That would make the code a bit harder to read. So, I would consider that as case of premature optimization that can be probably left for later if needed.